docs/overview/understanding.md

```diff
@@ -0,0 +1,328 @@
+---
+id: understanding
+title: "Understanding ZIO-Schema"
+---
+
+
+## Understanding ZIO-Schema
+
+ZIO-Schema is a part of ZIO that is used in many projects (e.g. ZIO-Flow, ZIO-SQL, etc.).
+ZIO is all about reification of your types. Reification means transforming something abstract (e.g. side effects, accessing fields, structure)  into something "real" (values).
+
+### Reification: functional effects
+In functional effects, we reify by turning side-effects into values.
+
+E.g. you might have a simple statement like
+```scala
+println("Hello")
+println("World")
+```
+and in ZIO we reify this statement to a value like
+
+```scala
+val effect1 = Task(println("Hello"))
+val effect2 = Task(println("World"))
+```
+
+and then are able to do awesome things like:
+```scala
+(Task(println("Hello")) zipPar Task(println("World"))).retryN(100)
+```
+
+### Reification: Optics
+In scala we have product types like this case class of a Person:
+```scala
+final case class Person(name: String, age: Int)
+```
+This case class has two fields:
+- a field "name" of type `String`
+- a field "age" of type `Int`
+
+The Scala language provides special support to access the fields inside case classes using the dot syntax:
+
+```scala
+val person = Person("Michelle", 32)
+val name = person.name
+val age  = person.age
+```
+
+However, this is a "special language feature", it's not "real" like the side effects we've seen in the previous example ( `println(..) vs. Task`println(...)))` ).
+
+Because these basic operations are not "real", we're unable to create an operator that we can use to
+e.g. combine two fields that are inside a nested structure.
+
+The solution to this kind of problem is called an "Optic". Optics provide a way to access the fields of a case class and nested structures.
+There are two main types of optics:
+- `Lens`: A lens is a way to access a field of a case class.
+- `Prims`: A prim is a way to access a field of a nested structure or a collection.
```

Comment:
Prims -> Prism. I would reword the description a bit. A prism is an optic that allows you to access an instance of a sealed trait as one of its concrete subtypes.